Update ESLint to 2.x

Hi,

currently available version of ESLint is 2.1.0. I think your dependencies should be updated. Also `index.js` should contain rules matching to those available now in the newest version of ESLint. Now, when I'm using `eslint-config-ember` with ESLint 2.1.0 I get that in test output: 
```bash
1:1  - Rule 'no-extra-strict' was removed and replaced by: strict (no-extra-strict)
1:1  - Rule 'no-wrap-func' was removed and replaced by: no-extra-parens (no-wrap-func)
1:1  - Rule 'spaced-line-comment' was removed and replaced by: spaced-comment (spaced-line-comment)
```

I can prepare PR if you want.

Regrads

I guess I spoke to soon. I'm testing out local versions of both `eslint-config-ember` and `ember-cli-eslint` using `eslint` @2.3.0, and when trying to `eslint-config-ember` inside of another app, I get this error when  I run `ember serve`:

```
Error: Cannot find module 'estraverse-fb'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at monkeypatch (/Users/bsipple/Development/ember-addons/ember-cli-eslint/node_modules/babel-eslint/index.js:59:32)
```

Apparently, this is related to `eslint 2.3.0` breaking a few things in `babel-eslint` -- the parser many might be using alongside `ember-cli-eslint`: https://github.com/babel/babel-eslint/issues/267

I was able to workaround it by installing eslint 2.2.0 as a devDependency in my app that was also trying to run my local `ember-cli-eslint`, (which, as mentioned has local symlinks to an `eslint-config-ember` project using `eslint 2.3.0`.

So... long story short, the error above is likely to be seen by anyone using our latest versions alongside `babel-eslint`. My suggestion might be to document the workaround of installing `eslint 2.2.0` directly next to where [`ember-cli-eslint` currently documents `babel-eslint` usage](https://github.com/jonathanKingston/ember-cli-eslint#eslint-babel) in the first place... at least until this behavior changes.

--- a/build.js
+++ b/build.js
@@ -1,3 +1,35 @@
 var fs = require('fs');
+var path = require('path');
 var config = require('./index');
-fs.writeFile("build/eslint.json", JSON.stringify(config), {encoding: 'utf8'});
+
+
+var dir = 'build';
+var jsonFile = 'eslint.json';
+var fullPath = path.join(dir, jsonFile);
+
+
+function justThrowErrors(err) { if(err) { throw err; } }
+
+
+function makeDir(dir, callback) {
+  fs.mkdir(dir, (err) => {
+    if (err) { throw err; }
+
+    callback();
+  });
+}
+
+
+function writeJson(jsonPath) {
+  fs.writeFile(jsonPath, JSON.stringify(config), { encoding: 'utf8' }, justThrowErrors);
+}
+
+
+// We can't write the file if we don't have the `build` directory.
+fs.stat(dir, (err) => {
+  if (err) {
+    makeDir(dir, () => writeJson(fullPath));
+  } else {
+    writeJson(fullPath);
+  }
+})

--- a/index.js
+++ b/index.js
@@ -2,23 +2,23 @@ var assign = require('object-assign');
 var config = require('eslint-config-nightmare-mode/browser');
 // Ember specific overrides
 var overrides = {
- "rules": {
-   "new-cap": [0,
+  "rules": {
+    "new-cap": [0,
       {
         "capIsNewExceptions": [
-            "Array",
-            "Boolean",
-            "Date",
-            "Error",
-            "Function",
-            "Number",
-            "Object",
-            "RegExp",
-            "String",
-            "Symbol",
-  
-            "Ember",
-            "A"
+          "Array",
+          "Boolean",
+          "Date",
+          "Error",
+          "Function",
+          "Number",
+          "Object",
+          "RegExp",
+          "String",
+          "Symbol",
+
+          "Ember",
+          "A"
         ]
       }
     ],

--- a/package.json
+++ b/package.json
@@ -22,12 +22,12 @@
   },
   "homepage": "https://github.com/jonathanKingston/eslint-config-ember#readme",
   "dependencies": {
-    "eslint-config-nightmare-mode": "^0.2.1",
+    "eslint-config-nightmare-mode": "^2.3.0",
     "object-assign": "^4.0.1"
   },
   "devDependencies": {
     "chai": "^2.3.0",
-    "eslint": "^1.4.1",
+    "eslint": "^2.4.0",
     "mocha": "^2.2.4"
   }
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-ember",
-  "version": "0.2.1",
+  "version": "0.3.0",
   "description": "Linting defaults to match Ember",
   "main": "index.js",
   "scripts": {

--- a/tests/fixtures/out/ember.js
+++ b/tests/fixtures/out/ember.js
@@ -2,10 +2,19 @@
   {
     "ruleId": "no-undef",
     "severity": 2,
-    "message": "\"Ember\" is not defined.",
+    "message": "'Ember' is not defined.",
     "line": 1,
     "column": 1,
     "nodeType": "Identifier",
+    "source": "Ember.A([1,2,3]);"
+  },
+  {
+    "ruleId": "no-magic-numbers",
+    "severity": 2,
+    "message": "No magic number: 1",
+    "line": 1,
+    "column": 10,
+    "nodeType": "Literal",
     "source": "Ember.A([1,2,3]);"
   },
   {
@@ -15,6 +24,22 @@
     "line": 1,
     "column": 11,
     "nodeType": "Punctuator",
+    "source": "Ember.A([1,2,3]);",
+    "fix": {
+      "range": [
+        11,
+        11
+      ],
+      "text": " "
+    }
+  },
+  {
+    "ruleId": "no-magic-numbers",
+    "severity": 2,
+    "message": "No magic number: 2",
+    "line": 1,
+    "column": 12,
+    "nodeType": "Literal",
     "source": "Ember.A([1,2,3]);"
   },
   {
@@ -24,6 +49,22 @@
     "line": 1,
     "column": 13,
     "nodeType": "Punctuator",
+    "source": "Ember.A([1,2,3]);",
+    "fix": {
+      "range": [
+        13,
+        13
+      ],
+      "text": " "
+    }
+  },
+  {
+    "ruleId": "no-magic-numbers",
+    "severity": 2,
+    "message": "No magic number: 3",
+    "line": 1,
+    "column": 14,
+    "nodeType": "Literal",
     "source": "Ember.A([1,2,3]);"
   }
 ]

--- a/tests/fixtures/out/index.js
+++ b/tests/fixtures/out/index.js
@@ -18,6 +18,15 @@
     "source": "var spoon = 'thhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhiiisssss iss longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg';"
   },
   {
+    "ruleId": "no-var",
+    "severity": 2,
+    "message": "Unexpected var, use let or const instead.",
+    "line": 1,
+    "column": 1,
+    "nodeType": "VariableDeclaration",
+    "source": "var spoon = 'thhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhiiisssss iss longgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg';"
+  },
+  {
     "ruleId": "no-console",
     "severity": 2,
     "message": "Unexpected console statement.",
@@ -29,10 +38,17 @@
   {
     "ruleId": "spaced-comment",
     "severity": 2,
-    "message": "Expected space or tab after // in comment.",
+    "message": "Expected space or tab after '//' in comment.",
     "line": 3,
     "column": 1,
     "nodeType": "Line",
-    "source": "//thig"
+    "source": "//thig",
+    "fix": {
+      "range": [
+        190,
+        190
+      ],
+      "text": " "
+    }
   }
 ]

